src/main/webapp/src/app/nav-bar/nav-bar.component.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { ComponentFixture, TestBed, inject, async } from '@angular/core/testing';
+import { NavBarComponent } from './nav-bar.component';
+import { Location } from '@angular/common';
+import { By } from '@angular/platform-browser';
+import { CommonModule } from '@angular/common';
+import { RouterTestingModule } from '@angular/router/testing';
+import { routes } from '../app-routing.module'
+
+describe('NavBarComponent', () => {
+ 
+  let component: NavBarComponent;
+  let fixture: ComponentFixture<NavBarComponent>;
+  let location: Location;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [ RouterTestingModule ],
+      declarations: [ NavBarComponent ],
+      providers: [
+        NavBarComponent
+      ]
+    })
+    .compileComponents();
+  });
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        CommonModule,
+        RouterTestingModule.withRoutes(routes)
+      ],
+    }),
+    
+    fixture = TestBed.createComponent(NavBarComponent);
+    component = fixture.componentInstance;
+    location = TestBed.get(Location);
+    fixture.detectChanges();
+  });
+
+  it('should create a NavBar', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should go to recommendation url', async(() => {
+    fixture.nativeElement.querySelector("#recommendation-navbar").click();
+    fixture.whenStable().then(() => {
+      expect(location.path()).toEqual('/recommendation');
+    });
+  }));
+  it('should go to home url', async(() => {
+    fixture.nativeElement.querySelector("#home-navbar").click();
+    fixture.whenStable().then(() => {
+      expect(location.path()).toEqual('/');
+    });
+  }));
+
+  it('recommendation route should be rendered correctly', () => {
+      let href = fixture.nativeElement.querySelector("#recommendation-navbar")
+      .getAttribute('href');
```

Comment:
I might not understand it properly, but what does the .getAttribute do? Because you are not storing it anywhere, right?  
(Link: https://developer.mozilla.org/en-US/docs/Web/API/Element/getAttribute)

Reply:
no it is not storing anything, but getAttribute takes the link from html file <div> container and checks if the link is that correct link , might not be the most useful test, but it just checks if the html file navigates to the right link